File: TestNodeJs/models/family.model.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Family extends Model {
    static associate(models) {
      this.hasMany(models.Person, {
        foreignKey: "family_id",
        as: "people",
      });

      this.belongsToMany(models.Asset, {
        through: "Family_Assets",
        foreignKey: "family_id",
        as: "assets",
        timestamps: false,
      });
    }
  }
  Family.init(
    {
      id: {
        type: DataTypes.INTEGER.UNSIGNED,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
      },
      kepala_keluarga: {
        type: DataTypes.STRING,
        allowNull: false,
      },
    },
    {
      sequelize,
      paranoid: true,
      modelName: "Family",
      tableName: "Families",
      createdAt: "created_at",
      updatedAt: "updated_at",
      deletedAt: "deleted_at",
    }
  );
  return Family;
};

File: TestGolang/models/family.model.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Family struct {
	Id              uint           `gorm:"primaryKey,autoIncrement" json:"id"`
	Kepala_keluarga string         `gorm:"not null" json:"kepala_keluarga"`
	Person          []*Person      `json:"anggota_keluarga"`
	Asset           []*Asset       `gorm:"many2many:family_assets"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"null" json:"deleted_at"`
}

func (Family) TableName() string {
	return "families"
}
